src/main/java/org/datavyu/plugins/javafx/JavaFxApplication.java

```diff
@@ -2,177 +2,127 @@
 
 import javafx.application.Application;
 import javafx.application.Platform;
-import javafx.beans.binding.Bindings;
-import javafx.beans.property.DoubleProperty;
+import javafx.concurrent.Task;
 import javafx.event.EventHandler;
-import javafx.scene.Scene;
 import javafx.scene.input.KeyCode;
 import javafx.scene.input.KeyEvent;
-import javafx.scene.layout.StackPane;
-import javafx.scene.media.Media;
-import javafx.scene.media.MediaPlayer;
-import javafx.scene.media.MediaView;
-import javafx.scene.paint.Color;
 import javafx.stage.Stage;
-import javafx.util.Duration;
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
+
 import org.datavyu.Datavyu;
-import org.datavyu.util.FrameRate;
+import org.datavyu.plugins.MediaPlayer;
+import org.datavyu.plugins.PlayerStateEvent;
 import org.datavyu.views.VideoController;
 
+import java.awt.*;
 import java.io.File;
+import java.net.URI;
 
-public class JfxApplication extends Application {
+public class JavaFxApplication extends Application {
```

Comment:
Could we simplify the whole thing by not using javafx?

Reply:
I will push a separate branch for this matter.